src/themes/elife/lib/dataProvider.ts

```diff
@@ -0,0 +1,69 @@
+import { first, text } from '../../../util'
+
+interface Response {
+  ok: boolean
+  articleData: { pdf: string; figuresPdf: string }
+}
+
+const normaliseWhitespace = (txt: string): string => {
+  return txt.replace(/\n/, ' ').replace(/ \s+|\n+/g, ' ')
+}
+
+const getNormalisedTextFromElement = (selector: string): string => {
+  const target = first(selector)
+  if (target !== null) {
+    const sourceText = text(target)
+    if (sourceText !== null) {
+      return normaliseWhitespace(sourceText)
+    }
+  }
+  return ''
+}
+
+const getPdfUrl = async (id: string, pdfType: string): Promise<string> => {
+  const allowedPdfTypes = ['article', 'figures']
+  if (!allowedPdfTypes.includes(pdfType)) {
+    return ''
+  }
+  const response = await query(id, window.fetch)
+  if (pdfType === 'figures') {
+    return Promise.resolve(response.articleData.figuresPdf)
+  }
+  return Promise.resolve(response.articleData.pdf)
+}
+
+export const getArticleId = (): string => {
+  return getNormalisedTextFromElement(
+    ':--identifier meta[content="https://registry.identifiers.org/registry/publisher-id"] ~ [itemprop="value"]'
+  )
+}
+
+export const getArticleDoi = (): string => {
+  return getNormalisedTextFromElement(
+    ':--identifier meta[content="https://registry.identifiers.org/registry/doi"] ~ [itemprop="value"]'
+  )
+}
+
+export const getArticleTitle = (): string => {
```

Comment:
What happens when the title contains markup, e.g. a superscript etc? Not tested. Addressing this would not be part of this PR, but mentioning as worthy of future investigation.

Reply:
Addressed in https://github.com/stencila/thema/pull/160/commits/db2f28c2a9c06be438aab30ecd3c7808bb67ba08#diff-8964265c04d747a981716c856aba1004R93